### ISSUE TITLE
bug: CoxPHFitter + spline baseline + strata + predict_cumulative_hazard 

### ISSUE THREAD

**Issue by iamdingkai** [CamDavidsonPilon/lifelines#1527]
Providing an error message, a bug, and how to fix it:

```
cox = CoxPHFitter(baseline_estimation_method = "spline", 
                                n_baseline_knots = 4,
                               )
cox.fit(df, 
            T_column, 
            E_column, 
            stata = strata_column)
cox.predict_cumulative_hazard(df_to_predict)
```

When df_to_predict.index.name != 'index', gets error message:
![image](https://github.com/CamDavidsonPilon/lifelines/assets/2110649/68d0dd72-2bf0-4b79-962a-52de1d2dcd45)

Reason for bug:
![image](https://github.com/CamDavidsonPilon/lifelines/assets/2110649/2745e7e2-fdac-48aa-8428-5b35b9ee368c)
![image](https://github.com/CamDavidsonPilon/lifelines/assets/2110649/e703d881-8c4a-4015-a8fd-2b329e86eb6c)

Basically, when strata is used, cox.predict_cumulative_hazard() resets index and assumes the index name is 'index'. After the reset_index, it assumes the index becomes a column 'index', and called this column later on. 
However, if df_to_predict.index.name != 'index', this will error out.
The way to fix this bug, is to store the index name of df_to_predict in a var, and use that var later on.
